src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.NativeAot.cs

```diff
@@ -32,26 +32,18 @@ public abstract partial class Array : ICollection, IEnumerable, IList, IStructur
         private int _numComponents;
 #pragma warning restore
 
-#if TARGET_64BIT
-        private const int POINTER_SIZE = 8;
-#else
-        private const int POINTER_SIZE = 4;
-#endif
-        //                                    Header       + m_pEEType    + _numComponents (with an optional padding)
-        private const int SZARRAY_BASE_SIZE = POINTER_SIZE + POINTER_SIZE + POINTER_SIZE;
-
         public int Length => checked((int)Unsafe.As<RawArrayData>(this).Length);
 
         // This could return a length greater than int.MaxValue
         internal nuint NativeLength => Unsafe.As<RawArrayData>(this).Length;
 
         public long LongLength => (long)NativeLength;
 
-        internal bool IsSzArray
+        internal unsafe bool IsSzArray
         {
             get
             {
-                return this.GetEETypePtr().BaseSize == SZARRAY_BASE_SIZE;
+                return this.GetMethodTable()->IsSzArray;
```

Comment:
This is used for small perf sensitive methods. `MethodTable.IsSzArray` looks more expensive than this condition. Can we fix `MethodTable.IsSzArray` to use this condition to avoid regressing perf?